### DISCUSSION TITLE
Keeping my fork current

### DISCUSSION THREAD

**Discussion by jsmorley** [microsoft/winget-pkgs#22586] [Q&A] [answered]
This may be a bit of an ignorant question, but I'm a tadpole in GitHub and hope someone can set me right.

The way I do changes to my manifest is to fetch / update a copy of a fork of the main repository on my GitHub account. Then I sync/update a copy of my fork that is on my local PC. I then make changes to my manifest, commit and push them back up to my fork, and do a pull-request to the main repository.

The question is, will there be a problem if there are changes to "other" manifests or code in general in the main repository while I'm in the middle of this process? Will it create a "conflict" in the pull-request that I, or someone else, will have to manage? Given the number of commits to the main repository that happen pretty often, I don't want to lose a "race" to get my changes in under the wire.

**Comment by ImJoakim** [comment, marked as answer]
If I'm understanding you properly (I might not). You are asking if you are planning on adding for example `Rainmeter Beta version 4.4.0.3508`. And the following happens:

1. You pull latest from master
2. You create your branch
3. The master branch gets modified while you're editing your branch/manifest for `Rainmeter Beta version 4.4.0.3508` (therefore not having the most recent changes)
4. You make a PR to add `Rainmeter Beta version 4.4.0.3508`

If this is the case, it will not cause any issues/conflicts as long as that specific version hasn't been modified.

**Reply by jsmorley** [reply]
Right. I just wanted to be sure that when I do a PR with my modified branch, which as you say I have added / changed a manifest like Rainmeter Beta version 4.4.0.3508, I don't cause any conflict with someone else's PR that has been committed in the meantime.

Thanks.

**Reply by ImJoakim** [reply]
Correct, you shouldn't encounter any issues/conflicts. If anything, every PR is being manually reviewed by a Community Moderator or a Windows Package Manager Administrator and they will help you if you ever encounter an issue with a PR.

**Reply by jsmorley** [reply]
Thank you. So far it has all gone great. Just wanted to be sure I'm being a good citizen...
